Azure/sonic-buildimage
# [Inventec][D6356] Kernel upgrade and few updates

**- Why I did it**

- Support Inventec D6356 with kernel 4.19
- fix xcvrd can't run issue after updated on master branch (cause by platform API)
- add auto find gpio base in swps driver to detect base changes when kernel changed

**- How I did it**
- Upgrade kernel to 4.19
- Update SFP/QSFP platform API base on latest xcvrd
- Add auto find gpio base in swps driver

**- How to verify it**
- Basic traffic tests
- Run "show interface status" "show interface transceiver presence" to check transceiver/port status
- Run "show platform fan", "show temperature", "show environment" ...etc to check device status
Please refer to the test report after these fixes for details.
[D6356_master.pdf](https://github.com/Azure/sonic-buildimage/files/5055102/D6356_master.pdf)


**- Which release branch to backport (provide reason below if selected)**




**- Description for the changelog**

- Upgrade kernel to 4.19
- Update SFP/QSFP platform API base on latest xcvrd
- Add auto find gpio base in swps driver

**- A picture of a cute animal (not mandatory but encouraged)**
